src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/transform/model/Transform.kt

```diff
@@ -0,0 +1,317 @@
+import com.amazon.opendistroforelasticsearch.indexmanagement.rollup.model.dimension.Terms
+import com.amazon.opendistroforelasticsearch.indexmanagement.util.IndexUtils
+import com.amazon.opendistroforelasticsearch.jobscheduler.spi.ScheduledJobParameter
+import com.amazon.opendistroforelasticsearch.jobscheduler.spi.schedule.CronSchedule
+import com.amazon.opendistroforelasticsearch.jobscheduler.spi.schedule.IntervalSchedule
+import com.amazon.opendistroforelasticsearch.jobscheduler.spi.schedule.Schedule
+import com.amazon.opendistroforelasticsearch.jobscheduler.spi.schedule.ScheduleParser
+import org.elasticsearch.common.bytes.BytesReference
+import org.elasticsearch.common.io.stream.StreamInput
+import org.elasticsearch.common.io.stream.StreamOutput
+import org.elasticsearch.common.io.stream.Writeable
+import org.elasticsearch.common.xcontent.LoggingDeprecationHandler
+import org.elasticsearch.common.xcontent.ToXContent
+import org.elasticsearch.common.xcontent.XContentBuilder
+import org.elasticsearch.common.xcontent.XContentFactory
+import org.elasticsearch.common.xcontent.XContentParser
+import org.elasticsearch.common.xcontent.XContentParser.Token
+import org.elasticsearch.common.xcontent.XContentParserUtils.ensureExpectedToken
+import org.elasticsearch.common.xcontent.XContentType
+import org.elasticsearch.index.query.AbstractQueryBuilder
+import org.elasticsearch.index.query.MatchAllQueryBuilder
+import org.elasticsearch.index.query.QueryBuilder
+import org.elasticsearch.index.seqno.SequenceNumbers
+import org.elasticsearch.search.aggregations.AggregatorFactories
+import java.io.IOException
+import java.time.Instant
+
+data class Transform(
+    val id: String = NO_ID,
+    val seqNo: Long = SequenceNumbers.UNASSIGNED_SEQ_NO,
+    val primaryTerm: Long = SequenceNumbers.UNASSIGNED_PRIMARY_TERM,
+    override val schemaVersion: Long,
+    override val jobSchedule: Schedule,
+    val metadataId: String?,
+    val updatedAt: Instant,
+    override val enabled: Boolean,
+    override val enabledAt: Instant?,
+    override val description: String,
+    val sourceIndex: String,
+    val dataSelectionQuery: QueryBuilder = MatchAllQueryBuilder(),
+    val targetIndex: String,
+    val roles: List<String>,
+    val pageSize: Int,
+    val groups: List<Dimension>,
+    val aggregations: AggregatorFactories.Builder
+) : ScheduleJob(schemaVersion = schemaVersion, jobSchedule = jobSchedule, enabled = enabled, enabledAt = enabledAt, description = description),
+    ScheduledJobParameter,
+    Writeable {
+
+    init {
+        aggregations.aggregatorFactories.forEach {
+            require(supportedAggregations.contains(it.type)) { "Unsupported aggregation [${it.type}]" }
+        }
+        when (jobSchedule) {
+            is CronSchedule -> {}
+            is IntervalSchedule -> {
+                require(jobSchedule.interval >= MINIMUM_JOB_INTERVAL) { "Transform job schedule interval must be greater than 0" }
```

Comment:
Hmm is this not already a requirement in job scheduler? If not I feel like this should be added there instead of here.

Reply:
sure, will add in the job scheduler if not present.